pxtblocks/code-validation/validateBlocksInSetExist.ts

```diff
@@ -0,0 +1,20 @@
+namespace pxt.blocks {
+    // validates that a combination of blocks in the set satisfies the required count
+    // returns the blocks that make the validator pass
+    export function validateBlocksInSetExist({ usedBlocks, blocks, count }: {
```

Comment:
Perhaps worth some discussion around whether or not we want to add a `requireUnique` flag here, which would make it so that each block in the set can only be counted once towards the desired total count. (i.e. instead of pushing all the blockInstances that match the id onto successfulBlocks, you push at most one, or zero if there are no matches).

Did you have any thoughts on this?

Reply:
The kinds of scenarios where this would be helpful for teacher-tool are very edge-case-y, since you can (usually) just do it with blocksExist and a threshold. The only scenarios I can think of where that is not the case would be if you wanted to check something like "two different kinds of loops" AND a third criteria that's not directly tied to that, all in the same criteria for some reason.

But it could be quite useful for tutorials, which don't have the same check/threshold architecture.

Reply:
Yeah that's reasonable. Some functionality, like requiring different counts for different blocks, is still done better with blocksExist, but it's possible that as this validator becomes more fleshed out, its functionality could become a superset of what blocksExist can do, then the blocksExist could head towards deprecation...but I wouldn't worry about that too much right now. For now, I think having the flag as an option but not using it in teacherTool is fine. If we want to use it later as a shorthand so we're not typing out a bunch of individual blocksExist checks, then that's great too. But no need for it to be a part of this change.

Reply:
Sounds good, I'll just add it as a parameter for the validator itself, then.